velox/exec/tests/utils/TpchQueryBuilder.cpp

```diff
@@ -0,0 +1,294 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/exec/tests/utils/TpchQueryBuilder.h"
+#include "velox/connectors/hive/HiveConnector.h"
+#include "velox/type/tests/FilterBuilder.h"
+#include "velox/type/tests/SubfieldFiltersBuilder.h"
+
+namespace facebook::velox::exec::test {
+
+static int64_t toDate(std::string_view stringDate) {
+  Date date;
+  parseTo(stringDate, date);
+  return date.days();
+}
+
+void TpchQueryBuilder::initialize(const std::string& dataPath) {
+  for (const auto& [tableName, columns] : kTables_) {
+    const fs::path tablePath{dataPath + "/" + tableName};
+    for (auto const& dirEntry : fs::directory_iterator{tablePath}) {
+      if (!dirEntry.is_regular_file()) {
+        continue;
+      }
+      // Ignore hidden files.
+      if (dirEntry.path().filename().c_str()[0] == '.') {
+        continue;
+      }
+      if (tableMetadata_[tableName].dataFiles.empty()) {
+        dwio::common::ReaderOptions readerOptions;
+        readerOptions.setFileFormat(format_);
+        std::unique_ptr<dwio::common::Reader> reader =
+            dwio::common::getReaderFactory(readerOptions.getFileFormat())
+                ->createReader(
+                    std::make_unique<dwio::common::FileInputStream>(
+                        dirEntry.path()),
+                    readerOptions);
+        tableMetadata_[tableName].type = reader->rowType();
+        const auto aliases = reader->rowType()->names();
+        // There can be extra columns in the file towards the end.
+        VELOX_CHECK_GE(aliases.size(), columns.size());
+        std::unordered_map<std::string, std::string> aliasMap(columns.size());
+        std::transform(
+            columns.begin(),
+            columns.end(),
+            aliases.begin(),
+            std::inserter(aliasMap, aliasMap.begin()),
+            [](std::string a, std::string b) { return std::make_pair(a, b); });
+        tableMetadata_[tableName].columnAliases = std::move(aliasMap);
+      }
+      tableMetadata_[tableName].dataFiles.push_back(dirEntry.path());
+    }
+  }
+}
+
+const std::vector<std::string>& TpchQueryBuilder::getTableNames() {
+  return kTableNames_;
+}
+
+TpchPlan TpchQueryBuilder::getQueryPlan(int queryId) const {
+  switch (queryId) {
+    case 1:
+      return getQ1Plan();
+    case 6:
+      return getQ6Plan();
+    default:
+      VELOX_NYI();
```

Comment:
nit: perhaps, add an error message

`VELOX_NYI*("TPC-H query {} is not supported yet", queryId);`